### PR TITLE
fix: [CDS-77402]: Improve nexus2 log message

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
In nexus2 if the tag is not found improve the error message from multiple tag found to no tag found.

Screenshot 2023-08-17 at 2 07 16 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51747)

### CODE REVIEW

File: 960-api-services/src/main/java/io/harness/nexus/service/NexusRegistryServiceImpl.java [hunk 1]
```diff
@@ -214,8 +221,14 @@ private BuildDetailsInternal getBuildNumber(NexusRequest nexusConfig, String rep
     List<BuildDetailsInternal> builds =
         getBuildDetails(nexusConfig, repository, port, artifactName, repositoryFormat, tag);
     builds = builds.stream().filter(build -> build.getNumber().equals(tag)).collect(Collectors.toList());
-
-    if (builds.size() == 1) {
+    if (builds.size() == 0) {
+      throw NestedExceptionUtils.hintWithExplanationException(
+          "Please check your Nexus repository for artifacts with same tag.",
```
[the remark this answers is not in the document]
Comment (sarthakkasat) [reply 960-api-services/src/main/java/io/harness/nexus/service/NexusRegistryServiceImpl.java:215]: Done

**Review by yogesh-chauhan — APPROVED** [review]

**Review by sourabh-awashti — COMMENTED** [review]
Can we write Junit for this change

**Review by sourabh-awashti — APPROVED** [review]

**Comment by sarthakkasat** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51747)* status: Some of the checks are too old. 1 file(s) to review. 1. unresolved discussions.

**Pull request merged**